packages/core/src/client/isSmartAccountClient.ts

```diff
@@ -22,9 +30,13 @@ export function isSmartAccountClient<
 >(
   client: Client<TTransport, TChain, TAccount>
 ): client is SmartAccountClient<TTransport, TChain, TAccount> {
-  return (
-    client && "middleware" in client && client.type === "SmartAccountClient"
```

Comment:
this check of the `type` was causing an E2E alchemy test to fail. So had to change this. The goal of the test is really to test what's going on on the right anyways: does this client support the actions we expect it to

Reply:
huh why was it failing? what was `client.type` referring to?

Reply:
ah because the AlchemyClient sets the type to `AlchemySmartAccountClient` which results in this not passing in the core methods. So instead I changed this to make sure the methods on it match the expectation instead